java-api/src/main/java/com/db/grad/javaapi/controller/BondController.java

```diff
@@ -29,5 +36,26 @@ public List<BondDto> getBonds() {
 
         return list;
     }
+
+
+    @GetMapping("/bonds/maturity/{date}")
+    public List<BondDto> getMaturityBonds(@PathVariable(value = "date") String date) {
+        List<BondDto> result = new ArrayList<>();
+
+        for (Bond bond : bondService.getAllBonds()) {
```

Comment:
Please move the search functionality inside bondService. In controller just convert result to DTO's